### PR TITLE
TST: use travis-ci container build

### PR DESCRIPTION
**Pull request by matthew-brett**
Switch to travis-ci containers:

http://docs.travis-ci.com/user/workers/container-based-infrastructure/ 
https://github.com/numpy/numpy/pull/6165

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,19 @@
 # for it to be on multiple physical lines, so long as you remember: - There
 # can't be any leading "-"s - All newlines will be removed, so use ";"s
 language: python
+
+# Run jobs on container-based infrastructure, can be overridden per job
+sudo: false
+
+cache:
+  directories:
+    - $HOME/.cache/pip
```
Comment (Eric89GXL) [inline .travis.yml]: caching doesn't work for open-source repos yet, right?

Comment (matthew-brett) [inline .travis.yml]: Is that right ? 

I have the impression from http://docs.travis-ci.com/user/workers/container-based-infrastructure/ that caching is available for public repositories using container-based infrastructure.

Comment (Eric89GXL) [inline .travis.yml]: Ahh, you're right. I remember seeing on a post somewhere that it was only for private ones, never mind!

**Comment by matthew-brett** [comment]
Any more comments here?  Otherwise I will merge soon.

**Comment by Eric89GXL** [comment]
Nope, LGTM

**Comment by agramfort** [comment]
LGTM

**Comment by matthew-brett** [comment]
Thanks - merging.

**Pull request merged**